CSS style 'filter: drop-shadow...' does not render correctly without repaint.

smfr opened the pull request:
#### 3210e3e92621749e154066495e94ccbcb6f6afa1
<pre>
CSS style &apos;filter: drop-shadow...&apos; does not render correctly without repaint.
<a href="https://bugs.webkit.org/show_bug.cgi?id=207586">https://bugs.webkit.org/show_bug.cgi?id=207586</a>
<a href="https://rdar.apple.com/59430460">rdar://59430460</a>

Reviewed by Alan Baradlay.

Visual overflow needs to account for CSS filters that move pixels, like `blur()`
and `drop-shadow()` in order for repaint to work. Fix this by having
`RenderBox::applyVisualEffectOverflow()` extend visual overflow for filters,
taking writing mode into account. We also need to trigger layout when visual
overflow due to filter outset changes happen, which is done in
`RenderStyle::changeAffectsVisualOverflow()`; `blur-change.html` tests this.

Since `RenderStyle::filterOutsets()` cannot provide an answer for reference filters,
remove it, and make `FilterOperations::outsets()` return a `std::optional` in this case,
which callers need to handle. Filter outset computation now needs to bottleneck
through `RenderBox::computeFilterOutsets()`.

This change does not correctly repaint descendants of an element with a filter;
a change like <a href="https://github.com/WebKit/WebKit/pull/12797">https://github.com/WebKit/WebKit/pull/12797</a> will be required
to fix this.

This change does also not take into account reference filters, which is tracked
in webkit.org/b/237538.

RenderLayer bounds are now inflated for filters, which results in composited layers
with composited filters getting inflated bounds; this was considered as a reasonable
trade-off for the complexity of recomputing visual overflow ignoring filters.

* LayoutTests/compositing/filters/sw-nested-shadow-overlaps-hw-nested-shadow-expected.txt:
* LayoutTests/compositing/geometry/bounds-with-filters-expected.txt: Added.
* LayoutTests/compositing/geometry/bounds-with-filters.html: Added.
* LayoutTests/css3/filters/drop-shadow-target-clipped.html:
* LayoutTests/fast/hidpi/filters-drop-shadow.html:
* LayoutTests/fast/repaint/blur-change-expected.txt: Added.
* LayoutTests/fast/repaint/blur-change.html: Added.
* LayoutTests/fast/repaint/drop-shadow-box-change-expected.txt: Added.
* LayoutTests/fast/repaint/drop-shadow-box-change.html: Added.
* LayoutTests/fast/repaint/drop-shadow-change-expected.txt: Added.
* LayoutTests/fast/repaint/drop-shadow-change.html: Added.
* LayoutTests/fast/repaint/drop-shadow-writing-modes-expected.txt: Added.
* LayoutTests/fast/repaint/drop-shadow-writing-modes.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-002.html:
* LayoutTests/platform/ios/TestExpectations: filters-drop-shadow-003.html is failing on bots,
but passes for me locally in sim and on hardware.
* LayoutTests/platform/win/compositing/geometry/bounds-with-filters-expected.txt: Added.
* LayoutTests/svg/filters/feMerge-nodes-unite-outsets-expected.html:
* LayoutTests/svg/filters/feMerge-nodes-unite-outsets.html: This test just seemed
to be wrong and failed after the repaint fix, so fix it.
* Source/WebCore/rendering/CSSFilterRenderer.cpp:
(WebCore::referenceFilterElement):
(WebCore::isIdentityReferenceFilter):
(WebCore::calculateReferenceFilterOutsets):
(WebCore::CSSFilterRenderer::isIdentity):
(WebCore::CSSFilterRenderer::calculateOutsets):
* Source/WebCore/rendering/CSSFilterRenderer.h:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::computeFilterOutsets const):
(WebCore::RenderBox::addVisualEffectOverflow):
(WebCore::convertOutsetsToOverflowCoordinates): Templatized because filter outsets are
RectEdges&lt;int&gt;.
(WebCore::RenderBox::applyVisualEffectOverflow const):
(WebCore::RenderBox::hasLayoutOverflow const):
* Source/WebCore/rendering/RenderBox.h:
(WebCore::RenderBox::applyVisualEffectOverflow):
* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::repaintOrMarkForLayout):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::calculateClipRects const):
* Source/WebCore/rendering/RenderLayer.h: hasAncestorWithFilterOutsets() can be private.
* Source/WebCore/style/StyleDifference.cpp:
* Source/WebCore/style/values/filter-effects/StyleFilter.cpp:
(WebCore::Style::Filter::calculateOutsets const):
* Source/WebCore/style/values/filter-effects/StyleFilter.h:

Canonical link: <a href="https://commits.webkit.org/313382@main">https://commits.webkit.org/313382@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8dc4b2d2fc49fc02195d1dad9a7c9f92989e59ef

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/162978 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/36457 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/29639 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/171916 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/119424 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/2779797a-f91b-4f5e-9d54-a1f7f51f68b5) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/164847 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/155/builds/36863 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/36459 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/126516 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/89119 "Build is in progress. Recent messages:Printed configuration; Running apply-patch; Checked out pull request; Skipped layout-tests; Running layout-tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/7f44f7ad-c53f-4498-bdea-379f6b3d1381) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/165937 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/155/builds/36863 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/146469 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/107092 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/0d90dbdb-cc30-4a25-9e75-64f6a9ed8d95) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/155/builds/36863 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/26451 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/19616 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/137646 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/24198 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/174420 "Built successfully") | | 
| | [❌ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/25107 "Found 1 new failure in wasm/WasmConstExprGenerator.cpp") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/25845 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/134827 "Passed tests") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/36128 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/30554 "Found 1 new test failure: imported/w3c/web-platform-tests/dom/events/Event-timestamp-high-resolution.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/134822 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/36357 "Built successfully and passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/36586 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/145994 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/98638 "Built successfully") | | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/164/builds/29833 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/22831 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/35624 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/106795 "Built successfully") | | | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/35123 "Hash 8dc4b2d2 for PR 45888 does not build (failure)") | [✅ 🧪 mac-site-isolation](https://ews-build.webkit.org/#/builders/185/builds/853 "Passed tests") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/35370 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/35271 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/832b63d7857dad5a5c90fcd868e380e927456015. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45888

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/106bbae255cb52f122d8792fe9033198ce5f57ab. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45888

smfr committed:
CSS style 'filter: drop-shadow...' does not render correctly without repaint.
https://bugs.webkit.org/show_bug.cgi?id=207586
rdar://59430460

Reviewed by NOBODY (OOPS!).

Visual overflow needs to account for CSS filters that move pixels, like `blur()`
and `drop-shadow()` in order for repaint to work. Fix this by having
`RenderBox::applyVisualEffectOverflow()` extend visual overflow for filters,
taking writing mode into account. We also need to trigger layout when visual
overflow due to filter outset changes happen, which is done in
`RenderStyle::changeAffectsVisualOverflow()`; `blur-change.html` tests this.

Since `RenderStyle::filterOutsets()` cannot provide an answer for reference filters,
remove it, and make `FilterOperations::outsets()` return a `std::optional` in this case,
which callers need to handle. Filter outset computation now needs to bottleneck
through `RenderBox::computeFilterOutsets()`.

This change does not correctly repaint descendants of an element with a filter;
a change like https://github.com/WebKit/WebKit/pull/12797 will be required
to fix this.

This change does also not take into account reference filters, which is tracked
in webkit.org/b/237538.

RenderLayer bounds are now inflated for filters, which results in composited layers
with composited filters getting inflated bounds; this was considered as a reasonable
trade-off for the complexity of recomputing visual overflow ignoring filters.

* LayoutTests/compositing/filters/sw-nested-shadow-overlaps-hw-nested-shadow-expected.txt:
* LayoutTests/compositing/geometry/bounds-with-filters-expected.txt: Added.
* LayoutTests/compositing/geometry/bounds-with-filters.html: Added.
* LayoutTests/css3/filters/drop-shadow-target-clipped.html:
* LayoutTests/fast/hidpi/filters-drop-shadow.html:
* LayoutTests/fast/repaint/blur-change-expected.txt: Added.
* LayoutTests/fast/repaint/blur-change.html: Added.
* LayoutTests/fast/repaint/drop-shadow-box-change-expected.txt: Added.
* LayoutTests/fast/repaint/drop-shadow-box-change.html: Added.
* LayoutTests/fast/repaint/drop-shadow-change-expected.txt: Added.
* LayoutTests/fast/repaint/drop-shadow-change.html: Added.
* LayoutTests/fast/repaint/drop-shadow-writing-modes-expected.txt: Added.
* LayoutTests/fast/repaint/drop-shadow-writing-modes.html: Added.
* LayoutTests/svg/filters/feMerge-nodes-unite-outsets-expected.html:
* LayoutTests/svg/filters/feMerge-nodes-unite-outsets.html: This test just seemed
to be wrong and failed after the repaint fix, so fix it.
* Source/WebCore/rendering/CSSFilterRenderer.cpp:
(WebCore::referenceFilterElement):
(WebCore::isIdentityReferenceFilter):
(WebCore::calculateReferenceFilterOutsets):
(WebCore::CSSFilterRenderer::isIdentity):
(WebCore::CSSFilterRenderer::calculateOutsets):
* Source/WebCore/rendering/CSSFilterRenderer.h:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::computeFilterOutsets const):
(WebCore::RenderBox::addVisualEffectOverflow):
(WebCore::convertOutsetsToOverflowCoordinates): Templatized because filter outsets are
RectEdges<int>.
(WebCore::RenderBox::applyVisualEffectOverflow const):
(WebCore::RenderBox::hasLayoutOverflow const):
* Source/WebCore/rendering/RenderBox.h:
(WebCore::RenderBox::applyVisualEffectOverflow):
* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::repaintOrMarkForLayout):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::calculateClipRects const):
* Source/WebCore/rendering/RenderLayer.h: hasAncestorWithFilterOutsets() can be private.
* Source/WebCore/style/StyleDifference.cpp:
* Source/WebCore/style/values/filter-effects/StyleFilter.cpp:
(WebCore::Style::Filter::calculateOutsets const):
* Source/WebCore/style/values/filter-effects/StyleFilter.h:

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/cf75dc3cc9128b35979c875e819443c97034c760. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45888

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8045cbbc9ee05e5a4f0ff8e22f5e8122887650e6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45888

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/3f9318ba8768f72041ff8c2537101b4c391a09db)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/154810 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/28069 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/21228 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/163570 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/108280 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/156683 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/28206 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/27918 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/119757 "Found 24 new test failures: compositing/filters/sw-layer-overlaps-hw-shadow.html compositing/filters/sw-nested-shadow-overlaps-hw-nested-shadow.html compositing/filters/sw-shadow-overlaps-hw-layer.html compositing/filters/sw-shadow-overlaps-hw-shadow.html compositing/layer-creation/overlap-transform-filter-contribution.html compositing/layer-creation/overlap-transform-filter.html css3/filters/composited-during-animation.html css3/filters/drop-shadow-current-color.html css3/filters/filter-animation-from-none-hw.html css3/filters/filter-animation-from-none-multi-hw.html ... (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/84669 "Passed tests") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/608a177b-d075-4183-97ab-9b17c337ad33) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/157769 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/22043 "Found 9 new test failures: compositing/filters/sw-layer-overlaps-hw-shadow.html compositing/filters/sw-nested-shadow-overlaps-hw-nested-shadow.html compositing/filters/sw-shadow-overlaps-hw-layer.html compositing/filters/sw-shadow-overlaps-hw-shadow.html compositing/layer-creation/overlap-transform-filter-contribution.html compositing/layer-creation/overlap-transform-filter.html compositing/masks/mask-and-drop-shadow.html css3/filters/drop-shadow-target-clipped.html fast/hidpi/filters-drop-shadow.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/139036 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/100450 "Passed tests") | | [❌ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/4dc0b5f9-9f35-493c-a217-cd7f0e1bbb07) 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/21128 "Found 2 new test failures: imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-002.html imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-003.html (failure)") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/19148 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/11396 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/130790 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/16880 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/166044 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/9334 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/18489 "Found 9 new test failures: compositing/filters/sw-layer-overlaps-hw-shadow.html compositing/filters/sw-nested-shadow-overlaps-hw-nested-shadow.html compositing/filters/sw-shadow-overlaps-hw-layer.html compositing/filters/sw-shadow-overlaps-hw-shadow.html compositing/layer-creation/overlap-transform-filter-contribution.html compositing/layer-creation/overlap-transform-filter.html compositing/masks/mask-and-drop-shadow.html fast/hidpi/filters-drop-shadow.html fast/repaint/blur-change.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/127859 "Found 26 new test failures: compositing/filters/sw-nested-shadow-overlaps-hw-nested-shadow.html compositing/filters/sw-shadow-overlaps-hw-layer.html compositing/filters/sw-shadow-overlaps-hw-shadow.html compositing/layer-creation/overlap-transform-filter-contribution.html compositing/layer-creation/overlap-transform-filter.html css3/filters/composited-during-animation.html css3/filters/drop-shadow-current-color.html css3/filters/filter-animation-from-none-hw.html css3/filters/filter-animation-from-none-multi-hw.html css3/filters/filter-animation-from-none-multi.html ... (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/27614 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/23188 "Found 23 new test failures: compositing/filters/sw-layer-overlaps-hw-shadow.html compositing/filters/sw-nested-shadow-overlaps-hw-nested-shadow.html compositing/filters/sw-shadow-overlaps-hw-layer.html compositing/filters/sw-shadow-overlaps-hw-shadow.html compositing/layer-creation/overlap-transform-filter-contribution.html compositing/layer-creation/overlap-transform-filter.html css3/filters/composited-during-animation.html css3/filters/filter-animation-from-none-hw.html css3/filters/filter-animation-from-none-multi-hw.html css3/filters/filter-animation-from-none-multi.html ... (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/127999 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/27538 "Built successfully") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/138673 "Exiting early after 10 failures. 134 tests run. 1 failures") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/84243 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/22890 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/15467 "Found 8 new test failures: compositing/filters/sw-layer-overlaps-hw-shadow.html compositing/filters/sw-nested-shadow-overlaps-hw-nested-shadow.html compositing/filters/sw-shadow-overlaps-hw-layer.html compositing/filters/sw-shadow-overlaps-hw-shadow.html compositing/layer-creation/overlap-transform-filter-contribution.html compositing/layer-creation/overlap-transform-filter.html fast/hidpi/filters-drop-shadow.html fast/repaint/blur-change.html (failure)") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/27230 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/91332 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/26808 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/27039 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/26881 "Built successfully") | | | | </details>

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/9f707172f434d2ca212c63533bc8fd1dd5f4c311)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/155675 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/28933 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/22092 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/164436 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/109489 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/f54ed00d-fe85-4bad-bfa2-2a5df89e1f64) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/157546 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/29080 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/28783 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/120486 "Found 3 new test failures: css3/filters/drop-shadow-current-color.html imported/w3c/web-platform-tests/css/filter-effects/css-filters-animation-drop-shadow.html platform/glib/damage/animations-blur.html (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/84937 "1 flakes 3 failures") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/2ddb912f-2230-44a3-88b0-882601aa34ae) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/158632 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/22661 "Found 4 new test failures: compositing/layer-creation/overlap-transform-filter-contribution.html compositing/masks/mask-and-drop-shadow.html css3/filters/drop-shadow-target-clipped.html fast/hidpi/filters-drop-shadow.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/139788 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/101175 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/5a2a0ed0-1228-4204-8121-7ff63635997d) 
| | [![loading-orange](https://github-production-user-asset-6210df.s3.amazonaws.com/3098702/291015173-08c448be-ac0a-4fd6-92a3-8165057445b7.png) 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/21747 "Build is in progress. Recent messages:OS: Tahoe (26.2), Xcode: 26.2; Running apply-patch; Checked out pull request; Exiting early after 10 failures. 13 tests run. 1 failures; Uploaded test results; 4 new passes 3 failures; 4 new passes 1 flakes 3 failures; Compiled WebKit (warnings); 4 new passes 1 flakes 1 failures; Found 2 new test failures: imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-002.html imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-003.html (failure)") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/19886 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/12266 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/131416 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/17620 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/166917 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/11091 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/19231 "Found 2 new test failures: compositing/masks/mask-and-drop-shadow.html fast/hidpi/filters-drop-shadow.html (failure)") | [![loading-orange](https://github-production-user-asset-6210df.s3.amazonaws.com/3098702/291015173-08c448be-ac0a-4fd6-92a3-8165057445b7.png) 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/128604 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Passed layout tests; 15 flakes 8 failures; Uploaded test results; 1 flakes 5 failures; Compiled WebKit; layout-tests; Found 5 new test failures: css3/filters/drop-shadow-current-color.html fast/hidpi/filters-drop-shadow.html imported/w3c/web-platform-tests/css/filter-effects/css-filters-animation-drop-shadow.html imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-002.html platform/glib/damage/animations-blur.html (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/28477 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/23920 "Found 1 new test failure: fast/hidpi/filters-drop-shadow.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/128736 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/28401 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/139413 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/86230 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/23551 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/16210 "Found 1 new test failure: fast/hidpi/filters-drop-shadow.html (failure)") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/28095 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/92198 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/27672 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/27902 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/27745 "Built successfully") | | | | </details>

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/0b72fdfd894ebf57d00644e40de732f45b60b36e)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/157188 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/30525 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/23717 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/166011 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/111270 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/159059 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/30660 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/30527 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/121734 "Found 1 new test failure: compositing/geometry/bounds-with-filters.html (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/85474 "3 failures") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/8e32a5c1-4209-4810-822e-3acc2b572169) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/160146 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/23979 "Found 3 new test failures: compositing/geometry/bounds-with-filters.html css3/filters/drop-shadow-target-clipped.html fast/hidpi/filters-drop-shadow.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/141144 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/102402 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/e6297d21-d8a2-40a1-96bc-2ab6c584e444) 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/23034 "Found 2 new test failures: imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-002.html imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-003.html (failure)") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/21273 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/13783 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/132714 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/18972 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/168496 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/12655 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/20592 "Found 2 new test failures: fast/hidpi/filters-drop-shadow.html imported/w3c/web-platform-tests/fetch/api/crashtests/huge-fetch.any.sharedworker.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/129866 "Found 3 new test failures: compositing/geometry/bounds-with-filters.html fast/hidpi/filters-drop-shadow.html imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-002.html (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/30126 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/25351 "Found 1 new test failure: fast/hidpi/filters-drop-shadow.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/129974 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/30049 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/140766 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/87870 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/24797 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/17570 "Found 1 new test failure: fast/hidpi/filters-drop-shadow.html (failure)") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/29760 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/93774 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/29282 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/29512 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/29409 "Built successfully") | | | | </details>

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/01ef7f0952a4f0ee73d5cd2fe5584f2469a33d19)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/157321 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/30658 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/23851 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/166145 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/111403 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/f1c4280a-f18f-4f40-a56f-3e9af00c3b20) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/159192 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/30793 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/30660 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/121844 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/85559 "1 flakes 1 failures") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/cb820d72-e337-47cb-b632-57b6c19a0524) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/160279 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/24090 "Found 2 new test failures: css3/filters/drop-shadow-target-clipped.html fast/hidpi/filters-drop-shadow.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/141267 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/102512 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/ca9c12f8-1435-4a80-aefc-60981970e05e) 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/23146 "Found 2 new test failures: imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-002.html imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-003.html (failure)") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/21392 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/13916 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/132825 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/19095 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/168630 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/12788 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/20715 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/129978 "Passed tests") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/30259 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/25470 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/130085 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/30182 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/140889 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/88093 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/24907 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/17694 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/29893 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/93907 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/29415 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/29645 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/29542 "Built successfully") | | | | </details>

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/b93b4cf53462f264e871a3f5466b2246843752b4)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/157482 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/30819 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/24012 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/166306 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/111564 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/07135552-7158-4e66-a9b6-36a7acc76c9a) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/159353 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/30954 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/30821 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/121959 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/85661 "1 failures") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/9e9ac599-bf1e-46e4-b3b1-cdd40014461a) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/160440 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/24233 "Found 1 new test failure: css3/filters/drop-shadow-target-clipped.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/141408 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/102628 "Passed tests") | | [❌ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/9a79bb43-f41e-4434-be37-a5ca2c815222) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/23289 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/21534 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/14077 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/132968 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/19236 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/168792 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/13098 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/20856 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/130106 "Found 1 new test failure: imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-002.html (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/30420 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/25612 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/130215 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/30343 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/141030 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/88284 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/25049 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/17835 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/30054 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/94255 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/29576 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/29806 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/29703 "Built successfully") | | | | </details>

smfr committed:
[iOS] nfl.com rubberbanding behaves incorrectly
https://bugs.webkit.org/show_bug.cgi?id=312813
rdar://170705188

Reviewed by NOBODY (OOPS!).

The changes in 254640@main added code that, if scrolling to 0,0, replaces the target zero position
with the current scroll offset, in an attempt to fix an issue in Mail on iPad. This code behaved
poorly when the user scrolls to the top with a rubberband, and the site does a
`window.scrollTo({ top: 0, behavior: 'smooth' })`, (which nfl.com does), which interrupts
any running scroll animation; it captured the currently negative Y offset as the static scroll position.

Fix by removing the changes in `_scrollToContentScrollPosition:...`. I could not reproduce the original
bug with these changes removed; it's possible that other OS changes fixed the `CGRectNull` issues that
254640@main was working around.

Test: fast/scrolling/ios/scroll-to-smooth-during-rubberband.html

* LayoutTests/fast/scrolling/ios/scroll-to-smooth-during-rubberband-expected.txt: Added.
* LayoutTests/fast/scrolling/ios/scroll-to-smooth-during-rubberband.html: Added.
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _scrollToContentScrollPosition:scrollOrigin:animated:interruptAnimation:]):

webkit-early-warning-system commented:
EWS run on current version of this PR (hash https://github.com/WebKit/WebKit/commit/72dd8f5a50d361c5fce93fe2a15f188a815cf008)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/157571 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/30908 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/24101 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/166395 "Built successfully") | [❌ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/111653 "Failed to checkout and rebase branch from PR 45888") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/8067532c-17d7-4d27-8c12-e855808a0e4c) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/159442 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/31043 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/30910 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/122018 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/59/builds/111653 "Failed to checkout and rebase branch from PR 45888") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/29c223ad-c773-4242-bf47-4bc11d73c251) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/160529 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/24300 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/141482 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/102687 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/b311aa6b-0ffc-4211-9dd9-e78bda1149df) 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/23356 "Found 2 new test failures: imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-002.html imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-003.html (failure)") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/21614 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/14166 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/133035 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/19308 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/168884 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/13296 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/20928 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/130178 "Found 1 new test failure: imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-002.html (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/30510 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/25691 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/130290 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/30432 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/141102 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/88430 "Failed to checkout and rebase branch from PR 45888") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/25104 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/17907 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/30143 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/94469 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/29665 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/29895 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/29792 "Built successfully") | | | | </details>

smfr committed:
CSS style 'filter: drop-shadow...' does not render correctly without repaint.
https://bugs.webkit.org/show_bug.cgi?id=207586
rdar://59430460

Reviewed by Alan Baradlay.

Visual overflow needs to account for CSS filters that move pixels, like `blur()`
and `drop-shadow()` in order for repaint to work. Fix this by having
`RenderBox::applyVisualEffectOverflow()` extend visual overflow for filters,
taking writing mode into account. We also need to trigger layout when visual
overflow due to filter outset changes happen, which is done in
`RenderStyle::changeAffectsVisualOverflow()`; `blur-change.html` tests this.

Since `RenderStyle::filterOutsets()` cannot provide an answer for reference filters,
remove it, and make `FilterOperations::outsets()` return a `std::optional` in this case,
which callers need to handle. Filter outset computation now needs to bottleneck
through `RenderBox::computeFilterOutsets()`.

This change does not correctly repaint descendants of an element with a filter;
a change like https://github.com/WebKit/WebKit/pull/12797 will be required
to fix this.

This change does also not take into account reference filters, which is tracked
in webkit.org/b/237538.

RenderLayer bounds are now inflated for filters, which results in composited layers
with composited filters getting inflated bounds; this was considered as a reasonable
trade-off for the complexity of recomputing visual overflow ignoring filters.

* LayoutTests/compositing/filters/sw-nested-shadow-overlaps-hw-nested-shadow-expected.txt:
* LayoutTests/compositing/geometry/bounds-with-filters-expected.txt: Added.
* LayoutTests/compositing/geometry/bounds-with-filters.html: Added.
* LayoutTests/css3/filters/drop-shadow-target-clipped.html:
* LayoutTests/fast/hidpi/filters-drop-shadow.html:
* LayoutTests/fast/repaint/blur-change-expected.txt: Added.
* LayoutTests/fast/repaint/blur-change.html: Added.
* LayoutTests/fast/repaint/drop-shadow-box-change-expected.txt: Added.
* LayoutTests/fast/repaint/drop-shadow-box-change.html: Added.
* LayoutTests/fast/repaint/drop-shadow-change-expected.txt: Added.
* LayoutTests/fast/repaint/drop-shadow-change.html: Added.
* LayoutTests/fast/repaint/drop-shadow-writing-modes-expected.txt: Added.
* LayoutTests/fast/repaint/drop-shadow-writing-modes.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-002.html:
* LayoutTests/platform/ios/TestExpectations: filters-drop-shadow-003.html is failing on bots,
but passes for me locally in sim and on hardware.
* LayoutTests/platform/win/compositing/geometry/bounds-with-filters-expected.txt: Added.
* LayoutTests/svg/filters/feMerge-nodes-unite-outsets-expected.html:
* LayoutTests/svg/filters/feMerge-nodes-unite-outsets.html: This test just seemed
to be wrong and failed after the repaint fix, so fix it.
* Source/WebCore/rendering/CSSFilterRenderer.cpp:
(WebCore::referenceFilterElement):
(WebCore::isIdentityReferenceFilter):
(WebCore::calculateReferenceFilterOutsets):
(WebCore::CSSFilterRenderer::isIdentity):
(WebCore::CSSFilterRenderer::calculateOutsets):
* Source/WebCore/rendering/CSSFilterRenderer.h:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::computeFilterOutsets const):
(WebCore::RenderBox::addVisualEffectOverflow):
(WebCore::convertOutsetsToOverflowCoordinates): Templatized because filter outsets are
RectEdges<int>.
(WebCore::RenderBox::applyVisualEffectOverflow const):
(WebCore::RenderBox::hasLayoutOverflow const):
* Source/WebCore/rendering/RenderBox.h:
(WebCore::RenderBox::applyVisualEffectOverflow):
* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::repaintOrMarkForLayout):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::calculateClipRects const):
* Source/WebCore/rendering/RenderLayer.h: hasAncestorWithFilterOutsets() can be private.
* Source/WebCore/style/StyleDifference.cpp:
* Source/WebCore/style/values/filter-effects/StyleFilter.cpp:
(WebCore::Style::Filter::calculateOutsets const):
* Source/WebCore/style/values/filter-effects/StyleFilter.h:

Canonical link: https://commits.webkit.org/313382@main
